src/mrpro/data/_KData.py

```diff
@@ -141,12 +142,17 @@ def reshape_acq_data(data):
                     setattr(current, subfield.name, reshape_acq_data(subcurrent))
 
         # Calculate trajectory and check if it matches the kdata shape
-        if isinstance(ktrajectory, KTrajectoryCalculator):
+        if isinstance(ktrajectory, KTrajectoryIsmrmrd):
```

Comment:
It would be nicer to have something like in line 156:
```
if ktraj is KTrajectoryRawShape:
      ktraj = ktraj.reshape(sort_idx, num_k2, num_k1)
```
but mypy did not like that.

Reply:
Also (orthogonal decision):
considering [#72](https://github.com/PTB-MR/mrpro/issues/72), it would make sense to first do the check if ktrajectory is an ISMRMRDCalculator (call with acquisitions and set ktrajectory as the result), a TrajectoryCalculator (call with header  and set ktrajectory as the result) or else. 
Then if  ktrajectory is Rawshape, reshape.

this would handle both cases where the trajectory is returned as raw shape in one reshaping block.